lib/truffle/rubygems/installing_extra_executables.rb

```diff
@@ -0,0 +1,38 @@
+module Gem
+  module InstallingExtraExecutables
+    extend self
+
+    def install_hooks_for(*extra_bin_dirs)
+      install_hook   = -> tool { hook(extra_bin_dirs, tool, :generate_bin) }
+      uninstall_hook = -> tool { hook(extra_bin_dirs, tool, :remove_executables, tool.spec) }
+      Gem.post_install_hooks.push(install_hook)
+      Gem.pre_uninstall_hooks.push(uninstall_hook)
+      [install_hook, uninstall_hook]
+    end
+
+    def remove_hooks(install_hook, uninstall_hook)
+      [Gem.post_install_hooks.delete(install_hook),
+       Gem.pre_uninstall_hooks.delete(uninstall_hook)]
+    end
```

Comment:
What calls this method?

Reply:
Nothing at the moment. I've planned to used it in tests, but did it was done differently in the end. However I think it's a good practice to also provide a way to remove the hooks not just add them.

Reply:
I think we should avoid dead code. It's unclear why this is here otherwise.
It doesn't seem useful for debugging either, as we can just not add the hooks.
Please remove it.

Reply:
I don't agree you don't remove code from a coherent library just because it's not used at the moment in the dependent project. This is in my opinion like that, a mini library which provides adding and removing of the hooks.

Reply:
But it's our own code or vendored code, so it makes sense to only keep what is used. What's the value of `remove_hooks` is it's never used?
I'm sure RubyGems has other places documenting how to remove a hook and it's not the role of this file.
I think it's better to keep our patches/additions/hooks minimal.

Reply:
I am not sure how does it relate that it's our code, we can structure things to libraries as well. The value is that it makes clear how to remove the extra-bin-dir hooks installed by this module if needed, not how to remove any hook. I don't see any technical issue with keeping it, it's a minor thing so as such I can decide not to accept your suggestion and keep the method. If you don't agree please contact me privately.

Reply:
I think having the analog `remove_hooks` makes sense. But, it will continue to trip up anyone trying to remove dead code. Maybe add a comment?

Reply:
I'll add a comment.